quinn-proto/src/endpoint.rs

```diff
@@ -490,24 +489,30 @@ impl Endpoint {
             version,
             retry_src_cid,
             orig_dst_cid,
-        };
-        if server_config.use_retry && !incoming.remote_address_validated() {
-            Some(DatagramEvent::Response(self.retry(incoming, buf)))
-        } else {
-            match self.accept(incoming, now, buf) {
-                Ok((ch, conn)) => Some(DatagramEvent::NewConnection(ch, conn)),
-                Err((_, response)) => response.map(DatagramEvent::Response),
-            }
-        }
+        }))
     }
 
     /// Attempt to accept this incoming connection (an error may still occur)
-    fn accept(
+    pub fn accept(
```

Comment:
The return type here (`Result<(ConnectionHandle, Connection), (ConnectionError, Option<Transmit>)>`) seems pretty elaborate. Can we do something a little more friendly?

Reply:
Will experiment with.

Reply:
Yeah I'm thinking just replacing `(ConnectionError, Option<Transmit>)` with:

```rs
/// Error type for attempting to accept an [`Incoming`]
#[derive(Debug)]
pub struct AcceptError {
    /// Underlying error describing reason for failure
    pub cause: ConnectionError,
    /// Optional response to transmit back
    pub response: Option<Transmit>,
}
```

Should be sufficient

Reply:
Did that.